chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,3105 @@
+  u8 chmode:4,
+     chpro:4;
+  u8 fixedpower:1,
+     fixedbandwidth:1,
+     TailElimination:2,
+     power:2,
+     bandwidth:2;
+  u8 rxfreq[4];
+  u8 txfreq[4];
+  u8 rxctcvaluetype:2,
+     rxctctypecode:1,
+     rev_1:1,
+     rxctchight:4;
+  u8 rxctclowvalue;
+  u8 rxsqlmode;
+  u8 txctcvaluetype:2,
+     txctctypecode:1,
+     rev_2:1,
+     txctchight:4;
+  u8 txctclowvalue;
+  u8 totPermissions:2,
+     tottime:6;
+  u8 vox:1,
+     companding:1,
+     scramble:1,
+     offlineorreversal:2,
+     rev_3:3;
+  u8 voxthreshold:4,
+     voxdelaytime:4;
+  u8 autoscan:1,
+     scanlist:7;
+  u8 alarmlist;
+  u8 dtmfsignalinglist:4,
+     pttidtype:2,
+     rev_4:2;
+  u16 dtmfcllid;
+  u8 reserve[3];
+} channels[1027];
+
+#seekto 0xb5c2;
+struct  {
+    u16 scannum;
+    u16 scanindex[16];
+} scandata;
+
+#seekto 0xb5e4;
+struct {
+   char name[14];
+   u8 scantxch:4,
+      scancondition:4;
+   u8 hangtime:4,
+      talkback:1,
+      scanstatus:3;
+   u8 chnum;
+   u16 specifych;
+   u16 PriorityCh1;
+   u16 PriorityCh2;
+   u8 scanmode;
+   u16 chindex[100];
+} scans[16];
+
+#seekto 0xc3e4;
+struct  {
+    u16 vfoscannum;
+    u16 vfoscanindex[3];
+} vfoscandata;
+
+#seekto 0xc3ec;
+struct {
+   u8 scantxch:4,
+      scancondition:4;
+   u8 hangtime:4,
+      talkback:1,
+      startcondition:1,
+      rev_1:2;
+   u8 scanmode;
+   u8 vhffreq_start[4];
+   u8 vhffreq_end[4];
+   u32 uhffreq_start;
+   u32 uhffreq_end;
+   u8 rev;
+} vfoscans[3];
+
+#seekto 0xc444;
+struct  {
+    u16 alarmnum;
+    u16 alarmindex[8];
+} alarmdata;
+
+#seekto 0xc456;
+struct {
+    char name[14];
+    u8 alarmtype:4,
+       alarmmode:4;
+    u16 jumpch;
+    u8 localalarm:1,
+       txbackground:1,
+       ctcmode:2,
+       rev_1:4;
+    u8 alarmtime:4,
+       alarmcycle:4;
+    u8 mictime:4,
+       txinterval:4;
+    u8 alarmid[8];
+    u8 alarmstatus;
+    u8 rev_3[1];
+    } alarms[8];
+
+#seekto 0xc848;
+struct  {
+    u16 dtmfnum;
+    u16 dtmfindex[4];
+} dtmfdata;
+
+#seekto 0xc852;
+struct {
+    u8 autoresettime:4,
+       codedelaytime:4;
+    u8 stunmode:2,
+       showani:1,
+       sidetone:1,
+       pttidtype:2,
+       rev_1:2;
+    char callid[10];
+    char stunid[10];
+    char revive[10];
+    char bot[16];
+    char eot[16];
+    char rev_2[8];
+} dtmfcomm;
+
+#seekto 0xc89a;
+struct {
+     char name[14];
+     u8 codelen:4,
+        signaling:4;
+     u8 groupcode:4,
+        intermediatecode:4;
+     char fastcall1[16];
+     char fastcall2[16];
+     char fastcall3[16];
+     char fastcall4[16];
+     char fastcall5[16];
+     char fastcall6[16];
+     char fastcall7[16];
+     char fastcall8[16];
+     char fastcall9[16];
+     char fastcall10[16];
+     u8 rev_1:6,
+        DecodingEnable:1,
+        EncodingEnable:1;
+     u8 dtmfstatus;
+     u8 rev_2[12];
+} dtmfs[4];
+"""
+
+
+class HandShakeStuts(Enum):
+    Normal = 0
+    Wrong = 1
+    PwdWrong = 3
+    RadioWrong = 4
+
+
+class Clone_TypeEnum(Enum):
```

Comment:
"CloneTypes"? Although these appear to be memory region definitions or something so maybe "MemoryRegions" would be better.